### PR TITLE
Update syntax for structs / mutable structs.

### PR DESCRIPTION
**Pull request by Nosferican**
immutable -> struct
type -> mutable struct

### CODE REVIEW

File: src/SortingAlgorithms.jl [hunk 1]
```diff
@@ -53,8 +53,8 @@ end
 uint_mapping(::ForwardOrdering, x::Float32)  = (y = reinterpret(Int32, x); reinterpret(UInt32, ifelse(y < 0, ~y, xor(y, typemin(Int32)))))
 uint_mapping(::ForwardOrdering, x::Float64)  = (y = reinterpret(Int64, x); reinterpret(UInt64, ifelse(y < 0, ~y, xor(y, typemin(Int64)))))
 
-uint_mapping{Fwd}(rev::ReverseOrdering{Fwd}, x) = ~uint_mapping(rev.fwd, x)
```
Comment (nalimilan) [inline src/SortingAlgorithms.jl]: `Fwd` isn't used at all. Below, `x::Real` would be enough.

Comment (Nosferican) [reply src/SortingAlgorithms.jl]: would it be
`uint_mapping(rev::ReverseOrdering{Fwd}, x::Real) = ~uint_mapping(rev.fwd, x)`?

Comment (nalimilan) [reply src/SortingAlgorithms.jl]: That would give:
```julia
uint_mapping(rev::ReverseOrdering, x) = ...
uint_mapping(::ReverseOrdering{ForwardOrdering}, x::Real) = ...
```

Comment (Nosferican) [reply src/SortingAlgorithms.jl]: Thanks!

Comment (nalimilan) [reply src/SortingAlgorithms.jl]: But your new commit doesn't implement that...

Comment (Nosferican) [reply src/SortingAlgorithms.jl]: You are correct. Fixing it now. Should the first line specify `x::Real` as well?

Comment (nalimilan) [reply src/SortingAlgorithms.jl]: I have no idea, but I don't see any reason to change that.

Comment (kmsquire) [reply src/SortingAlgorithms.jl]: It should not.  Although uncommon, the ordering object wrapped in the `ReverseOrdering` could potentially change the type of the elements being compared.

**Pull request merged**